itests/kit/node_opts.go

```diff
@@ -82,6 +84,13 @@ type OptBuilder func(activeNodes []*TestFullNode) node.Option
 // NodeOpt is a functional option for test nodes.
 type NodeOpt func(opts *nodeOpts) error
 
+func WithMockProofs() NodeOpt {
```

Comment:
there's already one on the ensemble itself, https://github.com/filecoin-project/lotus/blob/6302607650f9191bcc471547c595d712875b2f0f/itests/kit/ensemble_opts.go#L38

I think we should be able to fetch `n.options.mockProofs` when we instantiate `TestUnmanagedMiner` so there's no need for a new option, use what's there, just expose it in the `TestUnmanagedMiner` struct.

Reply:
I have removed tests with mock proofs completely for now. Will add it back as a separate test.

Reply:
I'm a little concerned about the complete removal of mock tests from this because there's so many places where you need to bypass ffi calls when you are running with mock proofs; I'll trust that you've got that under control but it is harder now the logic has been spread around a bit more.